### DISCUSSION TITLE
Limit length of stream: Difference between maxlen and max_records?

### DISCUSSION THREAD

**Discussion by Dronakurl** [ag2ai/faststream#1610] [Q&A] [answered]
```python
@broker.subscriber(
    stream=StreamSub("newimage", max_records=200, maxlen=100, group="mvp", consumer="2")
)
```
With this, I want to limit the size of my Stream in my `RedisBroker`. 
Does anyone know the difference between those two max-settings? I could not find it in the documentation.
I just want to prevent, that un-acknowledged messages fill up my harddrive indefinetely.

This is how I start my redis server
```bash
docker run -d --rm -p 6379:6379 --name redis redis 
```

**Comment by Lancetnik** [comment, marked as answer]
@Dronakurl hi again! Indeed, it is a confusing question
`max_records` is a subscriber option and represents how many messages with can read from stream by one request
`maxlen` is a publisher option to remove old messages at publishing if this limit exceeded

**Reply by theobouwman** [reply]
So if I only want to keep the last 20 messages, I need to set the `maxlen=20` and when the 21nd message gets added it automatically removed the oldest one?

**Reply by Lancetnik** [reply]
> So if I only want to keep the last 20 messages, I need to set the `maxlen=20` and when the 21nd message gets added it automatically removed the oldest one?

Yes, this is how Redis Stream works

**Reply by theobouwman** [reply]
Okay. And what if I push 200 messages with a maxlen=20. Will it first consume the 200 messages, and then keep to 20 most recent ones?

**Reply by Lancetnik** [reply]
> Okay. And what if I push 200 messages with a maxlen=20. Will it first consume the 200 messages, and then keep to 20 most recent ones?

Redis will remove overflowing messages regardless of whether they are read or not

**Reply by Lancetnik** [reply]
But, you should check Redis documentation instead of asking me, I am not so huge Redis expert, sorry
